naga/src/back/glsl/mod.rs

```diff
@@ -3722,6 +3722,49 @@ impl<'a, W: Write> Writer<'a, W> {
 
                         return Ok(());
                     }
+                    // FIXME: move to correct category
+                    Mf::QuantizeToF16 => match *ctx.resolve_type(arg, &self.module.types) {
+                        crate::TypeInner::Scalar { .. } => {
+                            write!(self.out, "unpackHalf2x16(packHalf2x16(vec2(")?;
+                            self.write_expr(arg, ctx)?;
+                            write!(self.out, "))).x")?;
+                            return Ok(());
+                        }
+                        crate::TypeInner::Vector {
+                            size: crate::VectorSize::Bi,
+                            ..
+                        } => {
+                            write!(self.out, "unpackHalf2x16(packHalf2x16(")?;
+                            self.write_expr(arg, ctx)?;
+                            write!(self.out, "))")?;
+                            return Ok(());
+                        }
+                        crate::TypeInner::Vector {
+                            size: crate::VectorSize::Tri,
+                            ..
+                        } => {
+                            write!(self.out, "vec3(unpackHalf2x16(packHalf2x16(")?;
+                            self.write_expr(arg, ctx)?;
```

Comment:
I don't love this bit, as we write `arg` twice, which might be undesirable if arg is a complex expression? It might be nicer to create a `quantizeToF16` function that can use a temporary variable. I see we do that for modf and frexp, however that's because they need a custom return type whereas this just returns a float/vec<n>

Reply:
We usually mark the `arg` as baked, that will make sure that the expression gets written to a temporary variable that we can then use multiple times. Example:

https://github.com/gfx-rs/wgpu/blob/6a60458790b70b584c5e5cfd534d63a884aa0294/naga/src/back/glsl/mod.rs#L1336

Reply:
Ah perfect that works great